format-common/src/main/java/io/cdap/plugin/format/connector/AbstractFileConnector.java

```diff
@@ -0,0 +1,186 @@
+    // no-op
+  }
+
+  @Override
+  public StructuredRecord transform(NullWritable key, StructuredRecord val) {
+    return val;
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext context,
+                                    ConnectorSpecRequest connectorSpecRequest) throws IOException {
+    ConnectorSpec.Builder builder = ConnectorSpec.builder();
+
+    String path = getFullPath(connectorSpecRequest.getPath());
+    ValidatingInputFormat inputFormat = getValidatingInputFormat(context, path);
+    // TODO: CDAP-18060 in 6.5 this will only have text and blob that will not access the file system,
+    //  to support other formats, we need to ensure get schema works
+    Schema schema = inputFormat.getSchema(new FormatContext(context.getFailureCollector(), null));
+    builder.setSchema(schema);
+    setConnectorSpec(connectorSpecRequest, builder);
+    return builder.build();
+  }
+
+  /**
+   * Return the full path including the scheme for the connector
+   */
+  protected String getFullPath(String path) {
+    return path;
+  }
+
+  /**
+   * Override this to provide any additional Configuration properties that are required by the FileSystem.
+   * For example, if the FileSystem requires setting properties for credentials, those should be returned by
+   * this method.
+   *
+   * @param path the path for the file system, some file system needs this path, for example, gcs needs the path to
+   *             determine the bucket name
+   */
+  protected Map<String, String> getFileSystemProperties(String path) {
+    return Collections.emptyMap();
+  }
+
+  /**
+   * Override this method to provide related plugins, properties or schema
+   *
+   * @param request the spec generation request
+   * @param builder the builder of the spec
+   */
+  protected void setConnectorSpec(ConnectorSpecRequest request, ConnectorSpec.Builder builder) {
+    // no-op
+  }
+
+  private ValidatingInputFormat getValidatingInputFormat(ConnectorContext context, String path) throws IOException {
+    String fileType = FileTypeDetector.detectFileType(path);
+    if (!FileTypeDetector.isSampleable(fileType)) {
+      throw new IllegalArgumentException(String.format("The given path %s is not isSamplable.", path));
+    }
+
+    String format = FileTypeDetector.detectFileFormat(fileType).name().toLowerCase();
+    PluginProperties.Builder builder = PluginProperties.builder();
+    builder.add("path", path);
+    builder.addAll(config.getProperties().getProperties());
+    builder.addAll(getFileSystemProperties(path));
+    ValidatingInputFormat inputFormat = context.getPluginConfigurer().usePlugin(
+      ValidatingInputFormat.PLUGIN_TYPE, format, UUID.randomUUID().toString(), builder.build());
+
+    if (inputFormat == null) {
+      throw new IOException(
+        String.format("Unsupported file format %s on path %s, the supported types are avro, blob, csv, delimited, " +
```

Comment:
this is technically not correct, since the formats are pluggable. I think it's ok to leave out the supported types in this case.